examples/Simple/Quicksort.hs

```diff
@@ -0,0 +1,61 @@
+{-# OPTIONS_GHC -Wno-name-shadowing #-}
+{-# OPTIONS_GHC -Wno-unused-matches #-}
+{-# LANGUAGE LinearTypes #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE NoImplicitPrelude #-}
+
+module Simple.Quicksort
+  ( testQuicksort
+  , quickSort
+  )
+where
+
+import GHC.Stack
+import qualified Data.Array.Mutable.Linear as Array
+import Data.Array.Mutable.Linear (Array)
+import Data.Unrestricted.Linear
+import Prelude.Linear hiding (partition)
+
+
+-- # Testing
+-------------------------------------------------------------------------------
+
+testQuicksort :: [Int] -> Bool
+testQuicksort xs = sort xs == quickSort xs
+
+
+-- # Quicksort
+-------------------------------------------------------------------------------
+
+quickSort :: [Int] -> [Int]
+quickSort xs = unur $ Array.fromList xs $ Array.toList . arrQuicksort
+
+arrQuicksort :: Array Int %1-> Array Int
+arrQuicksort arr = Array.size arr & \case
+  (Ur len, arr') -> go 0 (len-1) arr'
```

Comment:
Some places we call this `arr'`, some other we call this `arr1`. We should set a precedent somewhere.